spec/immutable-data-structures.html

```diff
@@ -6,27 +6,590 @@ <h1>Immutable Data Structures</h1>
+        </emu-note>
+      </emu-clause>
+      <emu-clause id="sec-append-record-field-to-list-functions">
+        <h1>AppendRecordFieldToList Functions</h1>
+        <p>An AppendRecordFieldToList function is an anonymous built-in function. When an AppendRecordFieldtoList function is called with arguments _key_ and _value_, the following steps are taken:</p>
+        <emu-alg>
+            1. Let _list_ be the *this* value.
+            1. Assert: Type(_list_) is List.
+            1. If Type(_value_) is Object, throw a *TypeError* exception.
+            1. Let _field_ be { [[Key]]: _key_, [[Value]]: _value_ }.
+            1. Append _field_ to the end of list _list_.
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-record.isrecord">
+        <h1>Record.isRecord ( _arg_ )</h1>
+        <p>The *isRecord* function takes one argument _arg_, and performs the following steps:</p>
+        <emu-alg>
+          1. Return ? IsRecord(_arg_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-record.prototype">
+        <h1>Record.prototype</h1>
+        <p>The initial value of *Record.prototype* is the value *null*.</p>
+
+        <p>This property has the attributes { [[Writable]]: false, [[Enumerable]]: false, [[Configurable]]: false }.</p>
       </emu-clause>
     </emu-clause>
   </emu-clause>
 
   <emu-clause id="sec-tuple-objects">
     <h1>Tuple Objects</h1>
-    <p>TBD</p>
     <emu-clause id="sec-tuple-constructor">
       <h1>The Tuple Constructor</h1>
-      <p>TBD</p>
-      <emu-clause id="sec-tuple-constructor-tuple-value">
-        <h1>Tuple ( _value_ )</h1>
-        <p>TBD</p>
+      <p>The Tuple constructor:</p>
+      <ul>
+        <li>is the intrinsic object <dfn>%Tuple%</dfn>.</li>
+        <li>is the initial value of the *"Tuple"* property of the global object.</li>
+        <li>creates and initializes a new Tuple object when called as a function.</li>
+        <li>is not intended to be used with the `new` operator or to be subclassed. It may be used as the value of an *extends* clause of a class definition but a *super* call to the Tuple constructor will cause an exception.</li>
+      </ul>
+      <emu-clause id="sec-tuple-items">
+        <h1>Tuple ( ..._items_ )</h1>
+        <p>When the `Tuple` function is called, the following steps are taken:</p>
+        <emu-alg>
+          1. If NewTarget is not *undefined*, throw a *TypeError* exception.
+          1. Let _elements_ be a List containing the arguments passed to this function.
+          1. Let _length_ to be the number of elements in _elements_.
+          1. For each element _e_ of _elements_,
+            1. If Type(_e_) is Object, throw a *TypeError* exception.
+          1. Let _tuple_ be a new Tuple value whose [[Sequence]] is _elements_.

```

Comment:
to use the common wording in the spec:
```suggestion
          1. Let _items_ be the list of arguments passed to this function.
          1. Let _length_ to be the number of elements in _items_.
          1. For each element _e_ of _items_,
            1. If Type(_e_) is Object, throw a *TypeError* exception.
          1. Let _tuple_ be a new Tuple value whose [[Sequence]] is _items_.
```